Code/GraphMol/Sgroup.h

```diff
@@ -0,0 +1,215 @@
+//
+//
+//  Copyright (C) 2002-2018 Greg Landrum and T5 Informatics GmbH
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+/*! \file Sgroup.h
+
+  \brief Defines the SGroup class
+
+*/
+#include <RDGeneral/export.h>
+#ifndef _RD_SGROUP_H
+#define _RD_SGROUP_H
+
+#include <unordered_map>
+
+#include <Geometry/point.h>
+#include <RDGeneral/types.h>
+#include <boost/smart_ptr.hpp>
+
+namespace RDKit {
+class ROMol;
+class Bond;
+class Atom;
+
+//! used to indicate errors from incorrect sgroup access
+class RDKIT_GRAPHMOL_EXPORT SGroupException : public std::runtime_error {
+ public:
+  //! construct with an error message
+  SGroupException(const char *msg) : std::runtime_error(msg){};
+  //! construct with an error message
+  SGroupException(const std::string &msg) : std::runtime_error(msg){};
+};
+
+class RDKIT_GRAPHMOL_EXPORT SGroup {
+ public:
+  enum class BondType {
+    XBOND,  // External
+    CBOND,  // Internal
+  };
+
+  typedef std::array<RDGeom::Point3D, 3> Bracket;
+
+  struct AttachPoint {
+    Atom *aAtom;
+    Atom *lvAtom;
+    std::string id;
+  };
+
+  struct CState {
+    Bond *bond;
+    boost::shared_ptr<RDGeom::Point3D> vector;
```

Comment:
Just use RDGeom::Point3D, we don't need a shared pointer here, it actually is much larger than three floats.

Reply:
Good point. I can't even remember why I wanted a shared pointer here.